pkg/plugins/policies/core/rules/rules.go

```diff
@@ -558,6 +569,27 @@ func toStringList(nodes []graph.Node) []string {
 	return rv
 }
 
+func mapToTargetRefs(tr common_api.TargetRef, gateways []*core_mesh.MeshGatewayResource) []common_api.TargetRef {
+	targetRefs := []common_api.TargetRef{}
+	switch tr.Kind {
+	case common_api.MeshGateway:
+		for _, gw := range gateways {
+			if gw.GetMeta().GetName() == tr.Name {
+				selectors := gw.Spec.GetSelectors()
```

Comment:
Should this include `gw.GetTags()`?

Reply:
I don't think we need them https://github.com/kumahq/kuma/blob/master/api/mesh/v1alpha1/gateway.pb.go#L156 based on the comment it means tags on the listener, so I think they are not used to build SNI.

Reply:
I think the `spec.tags` don't matter because if the selector matches, it can only be a `Dataplane` configured by the named MeshGateway, we don't need to check the MeshGateway tags.